### PR TITLE
 Add postCopyright into the default injects point

### PR DESCRIPTION
**Pull request by lengthmin**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes was maked (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [x] [Docs](https://github.com/theme-next/theme-next.org/tree/source/source/docs) in [NexT website](https://theme-next.org/docs/) have been added / updated (for features).

## PR Type
**What kind of change does this PR introduce?**

- [ ] Bugfix.
- [x] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

The default copyright style.

Issue resolved: N/A

## What is the new behavior?

you can create the copyright you like.

- Screenshots with this changes: 
after create custom copyright file:
image
- Link to demo site with this changes: https://lengthmin.me/posts/install-arch-wsl/

### How to use?
In NexT `_config.yml`:
```yml
#postCopyright: source/_data/post-copyright.swig
```
uncomment the line, and insert your own file.

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

**Comment by 1v9** [comment]
Cool, could you share your custom swig? We can add it to [Awesome-NexT](https://github.com/theme-next/awesome-next).

File: layout/_macro/post.swig [hunk 1]
```diff
@@ -345,6 +345,14 @@
       </div>
     {% endif %}
 
+    {% if not is_index %}
+      {% for inject_item in theme.injects.postCopyright %}
+        <div>
```
Comment (1v9) [inline layout/_macro/post.swig]: `<div><div>` is no needed I think

**Comment by jiangtj** [comment]
image
What is the difference between pointBodyEnd and postCopyright?
Maybe you can try adding postCopyright to pointBodyEnd.

**Comment by lengthmin** [comment]
> image
> What is the difference between pointBodyEnd and postCopyright?
> Maybe you can try adding postCopyright to pointBodyEnd.

their position are different, so I think it's better to add a new config.

File: layout/_macro/post.swig (hunk 1 above)
Comment (lengthmin) [reply layout/_macro/post.swig]: sure

**Comment by lengthmin** [comment]
> image
> What is the difference between pointBodyEnd and postCopyright?
> Maybe you can try adding postCopyright to pointBodyEnd.

and I think BodyEnd custom file is different from postCopyright custom file.
the BodyEnd behavior:
image

**Comment by jiangtj** [comment]
> their position are different, so I think it's better to add a new config.

If the wechat_subscriber and the reward are added through postBodyEnd, the two injection points overlap, so modify the implementation of the wechat_subscriber and the reward can be better, don't you think?

**Comment by lengthmin** [comment]
> Cool, could you share your custom swig? We can add it to [Awesome-NexT](https://github.com/theme-next/awesome-next).

I will share it later. Thanks for you appreciation! :satisfied:

**Comment by lengthmin** [comment]
> > their position are different, so I think it's better to add a new config.
> 
> If the wechat_subscriber and the reward are added through postBodyEnd, the two injection points overlap, so modify the implementation of the wechat_subscriber and the reward can be better, don't you think?

in my mind, I'd like to divide it into two files, because I have used `postBodyEnd` file to implement another func.
我觉得分成两个比较舒服... 因为我拿 `postBodyEnd` 当成了正文结束的提示。
如果不分我也是可以接受的，毕竟确实离得近了，调整下顺序的问题。

**Comment by jiangtj** [comment]
> in my mind, I'd like to divide it into two files, because I have used `postBodyEnd` file to implement another func.

image
你可以试试通过优先级确认前后位置。
You can try to confirm the position before and after by priority.

**Comment by jiangtj** [comment]
And the same injection point supports injecting multiple layouts, not just one. Just make sure the name of the injection is different.

**Comment by stevenjoezhang** [comment]
https://github.com/iissnan/hexo-theme-next/pull/1667

**Comment by lengthmin** [comment]
> [iissnan/hexo-theme-next#1667](https://github.com/iissnan/hexo-theme-next/pull/1667)

http://moxfive.xyz/2016/06/14/duoshuo-disqus-comment-count/#footer
我是从这`hexo-theme-yelee` 抄来的~ 

作者demo: http://moxfive.xyz/2016/06/14/duoshuo-disqus-comment-count/#footer

**Comment by lengthmin** [comment]
> Cool, could you share your custom swig? We can add it to [Awesome-NexT](https://github.com/theme-next/awesome-next).

https://gist.github.com/lengthmin/e5250848c6b2437a392d9d1411a0fbd6

**Comment by jiangtj** [comment]
Anyway, thanks for your time.  And pointBodyEnd and postCopyright are similar, I refactor in https://github.com/theme-next/hexo-theme-next/pull/980. (本来想过段时间再弄得QwQ，I thought about it for a while QwQ.)

**Comment by lengthmin** [comment]
Thanks to @jiangtj , This PR isn't elegant,
see https://github.com/theme-next/hexo-theme-next/pull/980#issuecomment-512229021

**Comment by jiangtj** [comment]
Looking forward to your PR In [Awesome NexT](https://github.com/theme-next/awesome-next)，for better post-copyright next plugin.

**Pull request closed without merging**